src/main/cpp/src/from_json_to_raw_map.cu

```diff
@@ -720,4 +722,57 @@ std::unique_ptr<cudf::column> from_json_to_raw_map(cudf::strings_column_view con
                                  mr);
 }
 
+std::pair<rmm::device_buffer, char> concat_json(cudf::column_view const& input,
+                                                rmm::cuda_stream_view stream,
+                                                rmm::device_async_resource_ref mr)
+{
+  CUDF_FUNC_RANGE();
+  auto const input_scv = cudf::strings_column_view{input};
+
+  auto constexpr num_levels = 256;
+  rmm::device_uvector<uint32_t> d_histogram(num_levels, stream);
+  thrust::fill(rmm::exec_policy(stream), d_histogram.begin(), d_histogram.end(), 0);
+  auto lower_level = std::numeric_limits<char>::min();
+  auto upper_level = std::numeric_limits<char>::max();
+
+  size_t temp_storage_bytes = 0;
+  cub::DeviceHistogram::HistogramEven(nullptr,
+                                      temp_storage_bytes,
+                                      input_scv.chars_begin(stream),
+                                      d_histogram.begin(),
+                                      num_levels,
+                                      lower_level,
+                                      upper_level,
+                                      input_scv.chars_size(stream),
+                                      stream.value());
+  rmm::device_buffer d_temp(temp_storage_bytes, stream);
+  cub::DeviceHistogram::HistogramEven(d_temp.data(),
+                                      temp_storage_bytes,
+                                      input_scv.chars_begin(stream),
+                                      d_histogram.begin(),
+                                      num_levels,
+                                      lower_level,
+                                      upper_level,
+                                      input_scv.chars_size(stream),
+                                      stream.value());
+  auto zero_level = d_histogram.begin() - lower_level;
+  auto first_non_zero_pos =
+    thrust::find(rmm::exec_policy(stream), zero_level + '\n', d_histogram.end(), 0);
```

Comment:
I see that this is finding number `0` in the range `[zero_level +  10, end())` (`[d_histogram.begin()  - 127 + 10, end())`.  Why is it called `first_non_zero_pos`?

Reply:
Here is finding a character that doesn't have any histogram count so maybe it should be called `first_zero_count_pos`?

Reply:
I am offsetting to `\n` so that if `\n` is not present, it's convenient to use `\n`.